src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -62,16 +64,18 @@
     private static final Logger LOGGER = Logger.getLogger(BaseAmazonWebServicesCredentials.class.getName());
 
     public static final int STS_CREDENTIALS_DURATION_SECONDS = 3600;
+
     private final String accessKey;
 
     private final Secret secretKey;
 
     private final String iamRoleArn;
     private final String iamMfaSerialNumber;
 
-    /**
-     * Old data bound constructor. It is maintained to keep binary compatibility with clients that were using it directly.
-     */
+    @CheckForNull
+    private Integer stsTokenDuration;
```

Comment:
This field should be protected somehow for concurrent access, as it is not final. Either make it `volatile` or guard it with a lock.